// react-autocomplete-searchbar/lib/components/AutocompleteDropdown.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import { array, string, func } from 'prop-types';

class AutocompleteDropdown extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "dropdownContent", () => {
      return this.props.suggestions.map(suggestion => React.createElement("a", {
        onClick: () => this.props.handleClick(suggestion),
        class: "ra-suggestion"
      }, suggestion[this.props.searchKey]));
    });
  }

  render() {
    return React.createElement("div", {
      class: "ra-dropdown"
    }, this.dropdownContent());
  }

}

_defineProperty(AutocompleteDropdown, "propTypes", {
  suggestions: array,
  searchKey: string,
  handleClick: func
});

_defineProperty(AutocompleteDropdown, "defaultProps", {
  suggestions: []
});

export default AutocompleteDropdown;
